### PR TITLE
RED GREEN REFACTOR

### PR DESCRIPTION
**Pull request by bcardarella**

### CODE REVIEW

File: source/posts/2016-03-23-the-thing-about-red-green-refactor-nobody-talks-about.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+More often than not you'll find this strategy to be very effective and
+you'll break through the wall faster than attempting to dream up the
+implementation in your head before writing the tests. You may also find
+that when you get to Step 5 the re-implemented code is much better than
+the spike you originally wrote.
+
+### You don't have to follow any "order"
+
+Take a look at the graph at the top of the page. Where does it start?
+Where does it end? Most people, because of the phrase "Red, Gree,
```
Comment (rwjblue) [inline source/posts/2016-03-23-the-thing-about-red-green-refactor-nobody-talks-about.md]: s/Gree/Green/

File: source/posts/2016-03-23-the-thing-about-red-green-refactor-nobody-talks-about.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+assume you inherit an application that has poor, or no, test coverage.
+It's perfectly OK to write simple acceptance tests that assert the
+current behavior of the application. Go directly to "Green". This is OK
+because the application is already live, its working. Once you have the
+happy paths covered in the acceptance tests you've written you can
+refactor with confidence.
+
+### Break the "rules"
+
+Like most rules "Red", "Green", "Refactor" makes sense for most cases, but
```
Comment (rwjblue) [inline source/posts/2016-03-23-the-thing-about-red-green-refactor-nobody-talks-about.md]: In some cases above the whole phrase is quoted, but here you quote each word.  Not sure which is right/wrong, but I think it should be internally consisten.

File: source/posts/2016-03-23-the-thing-about-red-green-refactor-nobody-talks-about.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+![](http://i.imgur.com/acmyARH.png)
+
+"Red", "Green", "Refactor"
+
+*Red*, *Green*, *Refactor*
+
+**RED** **GREEN** **REFACTOR**
+
+This process has been drilled into our head for over ten years. The goal
+is to push to to writing better software by ensuring that only what you
```
Comment (courajs) [inline source/posts/2016-03-23-the-thing-about-red-green-refactor-nobody-talks-about.md]: "push to to"

File: source/posts/2016-03-23-the-thing-about-red-green-refactor-nobody-talks-about.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+### You don't have to follow any "order"
+
+Take a look at the graph at the top of the page. Where does it start?
+Where does it end? Most people, because of the phrase "Red, Gree,
+Refactor" feel they *must* start with "Red". Well, you don't. Let's
+assume you inherit an application that has poor, or no, test coverage.
+It's perfectly OK to write simple acceptance tests that assert the
+current behavior of the application. Go directly to "Green". This is OK
+because the application is already live, its working. Once you have the
```
Comment (brzpegasus) [inline source/posts/2016-03-23-the-thing-about-red-green-refactor-nobody-talks-about.md]: `s/its/it's/`

**Comment by brzpegasus** [comment]
LGTM

File: source/posts/2016-03-23-the-thing-about-red-green-refactor-nobody-talks-about.md (hunk 2 above)
Comment (bcardarella) [inline source/posts/2016-03-23-the-thing-about-red-green-refactor-nobody-talks-about.md]: makes sense, thanks

**Pull request merged**